jinja2/filters.py

```diff
@@ -457,7 +457,9 @@ def do_indent(s, width=4, indentfirst=False):
             indent by two spaces and indent the first line too.
     """
     indention = u' ' * width
-    rv = (u'\n' + indention).join(s.splitlines())
+    lines = s.splitlines()
+    rv = lines.pop(0) + u'\n'
+    rv += u'\n'.join([indention + line if line else line for line in lines])
```

Comment:
@ThiefMaster well it's Generator Expressions vs. List Comprehension, I was reading and you are right for these case Generator Expression is better.
http://stackoverflow.com/questions/47789/generator-expressions-vs-list-comprehension